src/main/scala/mesosphere/marathon/core/task/termination/KillService.scala

```diff
@@ -50,4 +41,10 @@ trait KillService {
     * @param reason the reason why the task shall be killed.
     */
   def killUnknownTask(taskId: Task.Id, reason: KillReason): Unit
+
+  /**
+    * Begins watching immediately for terminated instances. Future is completed when all instances are reported in a
+    * terminal condition.
+    */
+  def watchForKilledInstances(instances: Seq[Instance])(implicit materializer: Materializer): Future[Done]
```

Comment:
I'm not particular fond of this design decision but it should be fine for now.

Reply:
I think we can improve that later...